### ISSUE TITLE
MongoConnectionException: Unknown Connection String Option 'persist'

### ISSUE THREAD

**Issue by ericcholis** [UnionOfRAD/lithium#1277]
I think the recent host/port refactor (668f310) had some unintended breaking changes with MongoDB. The `persist` option was deprecated in driver version 1.2.0, in favor of the driver handling it automatically. The [PHP documentation for MongoClient](http://php.net/manual/en/mongoclient.construct.php) shows that it can be included without any effect. However, it would appear that it is no longer a valid option, and throws a `MongoConnectionException`: "Found unknown connection string option 'persist' with value 'default'"

I've tested this behavior with the legacy MongoDB driver, versions 1.6.0, 1.6.9, 1.6.13, and 1.6.14 on PHP 5.6.22 and MongoDB 3.2.6. Each test threw a `MongoConnectionException`. Apparently, the PHP documentation is incorrect, `persist` cannot be included at all. There is an [old issue](https://jira.mongodb.org/browse/PHP-228) on the MongoDB JIRA that references this behavior, but appears to have no update.

The previous behaviour for `data\source\MongoDb` conditionally added the `persist` option if it was set in the connection string. Otherwise, it wasn't passed to the MongoClient constructor. See [line 308](https://github.com/UnionOfRAD/lithium/blob/1b3ec96f2ece20decf31c073af7a5ce8b4777153/data/source/MongoDb.php#L308).

```php
if ($persist = $cfg['persistent']) {
    $options['persist'] = $persist === true ? 'default' : $persist;
}
```

The new behavior provides a default `persistent` value of `false`, and overrides it with `default` if set to `true` in the connection string. Now, `persist` is always passed to the MongoClient constructor regardless of it's value, See [line 335](https://github.com/UnionOfRAD/lithium/blob/668f310cc897c80fb49bb49390df316b79d6cd04/data/source/MongoDb.php#L335)

```php
$options = array(
    'connect' => true,
    'connectTimeoutMS' => $this->_config['timeout'],
    'replicaSet' => $this->_config['replicaSet'],
    'persist' => $this->_config['persistent'] === true ? 'default' : $this->_config['persistent']
);
```

I wasn't sure how to approach a fix, given that removing `persist` entirely could be a breaking change. However, it's been removed since the 1.2.0 driver version.

**Comment by davidpersson** [comment]
Thanks for the helpful report. This wasn't intended... indeed. [1.2.0 was released in 2011](https://pecl.php.net/package/mongo): we probably should require a more recent version. Not sure what that should be, but at a minimum 1.2.0. So let's take the chance and remove support for passing a persistent configuration alltogether for li3 1.1.

**Comment by davidpersson** [comment]
If you like feel free to open a PR and remove any reference to persist/persistent in the MongoDb adapter. When doing so please also reference this ticket and the refactor commit. If you don't have time I can do this.

**Comment by davidpersson** [comment]
Fixed by e9160587e43cb5ec64b6b15373fb655002ec75c7
